docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+        │   │   └── [Module]Installer.php        
+        │   ├── [Module]BusinessFactory.php
+        │   ├── [Module]Facade.php
+        │   └── [Module]FacadeInterface.php
+        │
+        ├── Persistence
+        │   ├── [Module]EntityManagerInterface.php
+        │   ├── [Module]EntityManager.php
+        │   ├── [Module]QueryContainer.php
+        │   ├── [Module]PersistenceFactory.php
+        │   ├── [Module]RepositoryInterface.php
+        │   ├── [Module]Repository.php
+        │   └── Propel
+        │       ├── Abstract[TableName].php
+        │       ├── AbstractSpy[TableName]Query.php
+        │       ├── Mapper
+        │       │   └── [Module|Entity]Mapper.php
+        │       └── Schema
+        │           └── [organization_name]_[domain_name].schema.xml
+        │
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php          
+        │   ├── Facade
+        │   │   ├── [Module]To[Module]FacadeBridge.php
+        │   │   └── [Module]To[Module]FacadeInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php      
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php
+        │                
+        ├── [Module]Config.php
+        └── [Module]DependencyProvider.php
+```
+
+Used components
+- [Bridge](#bridge)
+- [Config](#module-configurations)
+- [Controller](#controller)
+- [Dependency Provider](#dependency-provider)
+- [Entity](#entity)
+- [Entity Manager](#entity-manager)
+- [Facade](#facade)
+- [Factory](#factory)
+- [Gateway Controller](#gateway-controller)
+- [Layout](#layout)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Models](#model)
+- [navigation.xml](#navigationxml)
+- [Plugin](#plugin), [Plugin Interface](#plugin-interface)
+- [Query Container](#query-container-facade)
+- [Query Object](#query-object)
+- [Repository](#repository)
+- [Schema](#persistence-schema)
+
+## Yves
+`Yves` application layer provides a lightweight Shop Application.
+
+```
+[Organization]
+└── Yves
+    └── [Module]
+        ├── Controller
+        │   ├── IndexController.php
+        │   └── [Name]Controller.php
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php                
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php           
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php
+        ├── Plugin
+        │   ├── Provider
+        │   │   ├── [Name]ControllerProvider.php
+        │   │   └── [Name]ServiceProvider.php
```

Comment:
what is this ServiceProvider controller?

Reply:
it's not a controller. as it is described a few lines below:
- **Service Provider** - Represents a cross-functional [Yves](#applications) service (eg: a validator).
- A `Service Provider` MUST implement `\Silex\ServiceProviderInterface and may extend \Spryker\Yves\Kernel\AbstractPlugin`.

Based on this, you can find 
- TwigMoneyServiceProvider
- TranslationServiceProvider